debezium-core/src/main/java/io/debezium/config/Configuration.java

```diff
@@ -2097,4 +2097,47 @@ default void forEachMatchingFieldName(Pattern regex, BiConsumer<String, String>
     default <T> void forEach(BiConsumer<String, String> function) {
         this.asMap().forEach(function);
     }
+
+    /**
+     * Returns the string config value from newProperty config field if it's set or its default value when it's not
+     * set/null.If both are null it returns the value of the oldProperty config field, or its default value when it's
+     * null.
+     * This fallback only works for newProperty fields that have a null / not-set default value!
+     *
+     * @param newProperty the new property config field
+     * @param oldProperty the old / fallback property config field
+     * @return the evaluated value
+     */
+    default String getFallbackStringProperty(Field newProperty, Field oldProperty) {
+        return getString(
+                newProperty,
+                () -> {
+                    LoggerFactory.getLogger(Configuration.class).warn("Using configuration property \"" + oldProperty.name()
+                            + "\" is deprecated and will be removed in future versions. Please use \"" + newProperty.name()
+                            + "\" instead.");
+                    return getString(oldProperty);
+                });
+    }
+
+    /**
+     * Returns the string config value of the provided Configuration from newProperty config field if it's set or its
+     * default value when it's not set/null.If both are null it returns the value of the oldProperty config field, or
+     * its default value when it's null.
+     * This fallback only works for newProperty fields that have a null / not-set default value!
+     *
+     * @param newProperty the new property config field
+     * @param oldProperty the old / fallback property config field
+     * @return the evaluated value
+     */
+    static String getFallbackStringProperty(Configuration config, Field newProperty, Field oldProperty) {
+        return config.getString(
+                newProperty,
+                () -> {
+                    LoggerFactory.getLogger(Configuration.class).warn("Using configuration property \"" + oldProperty.name()
```

Comment:
Let's better avoid getting a new logger instance for each call and put it into a static field as done elsewhere.

Reply:
Same in the other method.

Reply:
Where to get the logger from? I put it there and it's in prod only called when loading config.
This is an interface. The only other option would be a **public** CONFIGURATION_LOGGER accessible from everywhere. Are you fine with that?

Reply:
@rk3rn3r  The best approach would probably be an auxilliary `io.debezium.config.Loggers` with package protected members but I feel it would be a little bit of overengineering. I think `public static` attribute is acceptable in this case.

Reply:
Oh yes, interesting solution. But I agree, then I will revert to my previous version with the public member. thanks for the feedback @jpechane

Reply:
Done